auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackIDPPrompt.java

```diff
@@ -186,7 +194,32 @@ public void onComplete(@NonNull Task<AuthResult> task) {
             authResultTask
                     .addOnFailureListener(
                             new TaskFailureLogger(TAG, "Error linking with credential"))
-                    .addOnCompleteListener(new FinishListener());
+                    .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            FirebaseAuth auth = FirebaseAuth.getInstance();
+                            if (!task.isSuccessful() && task.getException() instanceof FirebaseAuthUserCollisionException) {
+                                setIntent(new Intent().putExtras(mActivityHelper.getMergeFailedIntent()));
```

Comment:
The new version is a little bit different, but I'm basically doing the same thing as before.
If the task isn't successful, it means the link failed so I need to delete the anonymous user and get the old uid. `delete()` also signs out the user so I then need to reauthenticate the user and pass on the intent containing the old uid to be manually merged. This scenario only happens when the Google/Facebook/Twitter etc. account already exists.